log4j-1.2-api/src/main/java/org/apache/log4j/config/Log4j1ConfigurationParser.java

```diff
@@ -159,7 +159,7 @@ private void warn(final String string) {
     }
 
     private void buildAppender(final String appenderName, final String appenderClass) {
-        switch (appenderClass) {
+        switch (appenderClass.trim()) {
```

Comment:
@garydgregory: I double checked that this private method is never called with a null `appenderClass`, but can you check again?